lib/components/switch.ts

```diff
@@ -1,9 +1,47 @@
-import { commonComponentProps } from "lib/common/layout"
+import {
+  type CommonComponentProps,
+  commonComponentProps,
+} from "lib/common/layout"
+import { expectTypesMatch } from "lib/typecheck"
+
 import { z } from "zod"
 
-export const switchProps = commonComponentProps.extend({
-  ftype: z.literal("switch"),
-  switchType: z.enum(["spst"]).default("spst"),
-  isNormallyClosed: z.boolean().default(false),
-})
-export type SwitchProps = z.input<typeof switchProps>
+export interface SwitchProps extends CommonComponentProps {
+  ftype: "switch"
+  type?: "spst" | "spdt" | "dpst" | "dpdt"
+  isNormallyClosed?: boolean
+  spdt?: boolean
+  spst?: boolean
+  dpst?: boolean
+  dpdt?: boolean
+}
+
+export const switchProps = commonComponentProps
+  .extend({
+    ftype: z.literal("switch"),
+    type: z.enum(["spst", "spdt", "dpst", "dpdt"]).optional(),
+    isNormallyClosed: z.boolean().optional().default(false),
+  })
+  .transform((val) => {
```

Comment:
`val` is a "banned name" there's always a better name than val https://github.com/tscircuit/handbook/blob/main/guides/code.md

try `props`